ovirt_sriov.py

```diff
@@ -0,0 +1,384 @@
+                number_of_virtual_functions=vfs,
+            )
+            nic_service.update_virtual_functions_configuration(virtual_functions_configuration=vf_config)
+            changed = True
+
+        if changed:
+            self.changed = True
+
+
+def main():
+    # Module Args
+    argument_spec = ovirt_full_argument_spec(
+        name=dict(aliases=['host'], required=True),
+        interface=dict(required=True),
+        vfs=dict(default=None, type='int'),
+        allowed_networks=dict(
+            choices=['all', 'specific'],
+            default=None,
+        ),
+        networks=dict(default=None, type='list'),
+        labels=dict(default=None, type='list'),
+    )
+
+    # Create & check module
+    module = AnsibleModule(
+        argument_spec=argument_spec, 
+        supports_check_mode=True,
+    )
+    
+    check_sdk(module)
+    
+    try:
+        # Connection to oVirt
+        auth = module.params.pop('auth')
+        connection = create_connection(auth)
+        hosts_service = connection.system_service().hosts_service()
+        sriov_module = SRIOVModule(
+                connection=connection,
+                module=module,
+                service=hosts_service,
+        )
+
+        # Verify host exists
+        host = sriov_module.search_entity()
+        if host is None:
+            raise Exception("Host '%s' was not found." % module.params['name'])
+
+        # Params
+        interface = module.params['interface']
+        vfs = module.params['vfs']
+        allowed_networks = module.params['allowed_networks']
+        networks = module.params['networks']
+        labels = module.params['labels']
+        
+        # Host services
+        hosts_service = connection.system_service().hosts_service()
+        networks_service = connection.system_service().networks_service()
+        host_service = hosts_service.host_service(host.id)
+        nics_service = host_service.nics_service()
+
+        # Verify interface exists
+        nic = search_by_name(nics_service, interface)
+        if nic is None:
+            raise Exception("Interface '%s' was not found." % module.params['interface'])
+        else:
+            nic_service = nics_service.nic_service(nic.id)
+
+        # Verify given networks exist
+        if networks:
+            for network in networks:
+                if search_by_name(networks_service, network) is None:
+                    raise Exception("Network '%s' was not found." % network)
+
+        # Check for and apply updates
+        if sriov_module.has_updates(networks_service, nics_service, nic_service):
+            if allowed_networks:
+                sriov_module.update_vf_allowed_networks(nics_service, nic_service)
+                if allowed_networks == 'specific':
+                    sriov_module.update_vf_networks(networks_service, nic_service)
+                    if labels:
+                        sriov_module.update_vf_labels(nic_service)

```

Comment:
I'm not sure, but I think the way this is written if update_vf_labels marks 'changed' as false, it will override a mark of true by the update_vf_networks and update_vf_allowed_networks.  Same is said for update_vf_networks overriding update_vf_allowed_networks.  If it's changed later it will take president.  Just need to ensure that if nothing is going to change in these nests that it doesn't hit them.

Reply:
I don't update the module's 'changed' variable unless the local changed variable is true at the end of each function. It's set false by default so I'm only updating it if it's true.